zipkin-receiver-kafka/src/main/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessor.scala

```diff
@@ -1,41 +1,44 @@
 package com.twitter.zipkin.receiver.kafka
 
 import com.twitter.zipkin.thriftscala.{Span => ThriftSpan}
-import kafka.consumer.{Consumer, ConsumerConnector, ConsumerConfig}
-import kafka.serializer.Decoder
 import com.twitter.util.{Closable, CloseAwaitably, FuturePool, Future, Time}
 import java.util.concurrent.{TimeUnit, Executors}
 import java.util.Properties
-import java.net.{SocketAddress, InetSocketAddress}
-
+import kafka.consumer.{Consumer, ConsumerConnector, ConsumerConfig}
+import kafka.serializer.{Decoder, StringDecoder}
 import com.twitter.scrooge.BinaryThriftStructSerializer
 import com.twitter.zipkin.conversions.thrift.{thriftSpanToSpan, spanToThriftSpan}
 import kafka.message.Message
 
+
 object KafkaProcessor {
 
   type KafkaDecoder = Decoder[Option[List[ThriftSpan]]]
 
-  def apply(
+  val defaultKeyDecoder = new StringDecoder()
+
+  def apply[T](
     topics:Map[String, Int],
     config: Properties,
     process: Seq[ThriftSpan] => Future[Unit],
-    decoder: KafkaProcessor.KafkaDecoder
-  ): KafkaProcessor = new KafkaProcessor(topics, config, process, decoder)
+    keyDecoder: Decoder[T],
+    valueDecoder: KafkaDecoder
+  ): KafkaProcessor[T] = new KafkaProcessor(topics, config, process, keyDecoder, valueDecoder)
 }
 
-class KafkaProcessor(
+class KafkaProcessor[T](
```

Comment:
Did you choose 'T' because other code was using the java-style generic 'T' or was this your choice? In Scala, we tend to start with 'A' and then go alphabetically: http://docs.scala-lang.org/style/naming-conventions.html#type_parameters_generics